installers/macos/preinstall

```diff
@@ -27,5 +34,76 @@ then
 	$INSTALL_LOCATION/scripts/service_stop
 fi
 
-exit 0
+# We need to ensure that there is a psuedo user (aka role) account for
+# our daemon, so that the daemon runs with minimal permissions and not
+# as the usual logged in user.  This is similar to the "nobody"
+# account.
+
+dscl . read /Users/$USERNAME >/dev/null 2>/dev/null
+if [ $? -eq 0 ]
+then
+	echo "Using existing pseudo user $USERNAME"
+	exit 0
+fi
+
+# We need to create a new pseudo user using a "role account".  These
+# have UIDs in the 200-400 range.
+#
+# Apple does not give us any tools to find or request an unused UID,
+# so we have to search for an unused UID.
+#
+# Find all of the already used role account UIDs and exclude them from
+# the allowable range. Build the free set backwards, so that we're not
+# as likely to collide with well-known services (that may not
+# currently be installed).
+
+USED_IDS=$(dscl . list /Users UniqueID | awk '{ if ($2 >= 200 && $2 < 400) print $2}' | sort -n)
+FREE_IDS=""
+for k in {200..399}
+do
+	echo $USED_IDS | grep -v -q $k && FREE_IDS="$k $FREE_IDS"
+done
+
+# We probably only need to try to create the pseudo user using the
+# first free UID in our list, but lets try others if we get an error.
+#
+# BTW `sysadminctl -addUser ...` does not set an exit code if our UID
+# guess is already in use, so we need to check the user database to
+# see if it was actually created.
+#
+# However, 'sysadminctl' does set an exit code if we are not running
+# as root.
+#
+# NEEDSWORK: The value of the GID is another mess. There is another
+# list/database of groups and group memberships and their numeric IDs
+# are unrelated to the UID numbers.  And I couldn't find a similar
+# `sysadminctl` command for creating a group; from what I could tell
+# you still have to use 5 or 6 `dscl . create /Group ...` commands.
+# I'm not going to bother with that. We only need the pseudo user ID
+# so that the daemon will run with low privileges and we only need to
+# do a `chown` on a few files in the installation directory; we don't
+# actually need a specific group.
+
+for k in $FREE_IDS
+do
+	echo "Trying to create pseudo user $USERNAME with UID $k."
+
+	sysadminctl -addUser $USERNAME -fullName "Git OTEL Telemetry" -UID $k -GID $k -roleAccount
+	RESULT=$?
+	if [ $RESULT -ne 0 ]
+	then
+		echo "Could not create a pseudo user for $USERNAME"
+		exit $RESULT
+	fi
+
+	dscl . read /Users/$USERNAME >/dev/null 2>/dev/null
+	RESULT=$?
+	if [ $RESULT -eq 0 ]
+	then
+		echo "Pseudo user $USERNAME successfully created with UID $k."
+		exit 0
+	fi
+done
 
+echo "Could not create a pseudo user for $USERNAME"
```

Comment:
We'd also end up here if there are no free IDs left? Something worth calling out explicitly in that case? If `FREE_IDS` was empty?